pkg/utils/helpers.go

```diff
@@ -93,15 +101,53 @@ func RemoveKey(list []string, key string) []string {
 	return newList
 }
 
+func CheckMapKeys(razeeConfigValues map[string]string, referenceList []string) []string {
+	missingItems := []string{}
+	for _, referenceItem := range referenceList {
+		if _, exists := razeeConfigValues[referenceItem]; !exists {
+			missingItems = append(missingItems, referenceItem)
+		}
+	}
+	return missingItems
+}
+
 func RetrieveSecretField(in []byte) (string, error) {
 	decodedString := b64.StdEncoding.EncodeToString(in)
 	decoded, err := b64.StdEncoding.DecodeString(decodedString)
 
 	return strings.Trim(string(decoded), " \r\n"), err
 }
 
+func ExtractCredKey(secret *corev1.Secret, sel corev1.SecretKeySelector) ([]byte, error) {
+	var value []byte
+	var error error
+	if value, ok := secret.Data[sel.Key]; ok {
+		return value, nil
+	} else if !ok {
+		error = fmt.Errorf("secret %s key %q not in secret", sel.Key, sel.Name)
+	}
+
+	return value, error
+}
+
+func GetDataFromRhmSecret(request reconcile.Request, sel corev1.SecretKeySelector, client client.Client) (error, []byte) {
```

Comment:
We can make this generic. You've hard coded the secret name. Let's pass in a name and namespace and have it return values.

Reply:
@zach-source can you confirm you want to use this instead of the function in razeedeployment_controller ? this is a dup.

Reply:
Just use the function in razeedeployment and let's clean up dupes/unused code.